Refactor shaders (separate chunks and main)

dmnsgn opened the pull request:
Notable changes:
- code moved to shaders (to make the separation evident and git history more readable)
- add folders:
  - `chunks`: contains variable and methods declaration to be imported in main shaders. They are all exported in `shaders/index.js`
  - `pipeline`: all the necessary pass for the rendering pipeline (Depth, Material and Overlay - should this be moved to post-pro?)
  - `post-processing`: pretty self explanatory
  - other main shader (reflection-probe, skybox and error)
- nice `/* glsl */` for template string highlight

Modifications along the way:
- Use same baseColor and alpha shader chunks for the rendering pipeline (check for `#define DEPTH_PASS_ONLY` and `DEPTH_PASS_ONLY`)
- Anywhere there is a `#ifdef`, move common uniforms up (example: `uEmissiveColor`)
- Fix double side normal in DepthPrePass, use  `float(gl_FrontFacing) * 2.0 - 1.0;`
- Typo `getBaseColorAndMetallicRoughnessFromSpecularGlossiness`
- Use `SHADERS.octMap` in `reflection-probe` (difference: is added EPSILON to envMapOctahedral(vec3 dir))
- Use math PI and TWO_PI for envMapEquirect (difference: greater precision)
- Refactor irradiance:
  - removed unused eyeDirWorld in vec3 getIrradiance(vec3 eyeDirWorld, vec3 normalWorld)
  - add map and encoding as input (eg make it work for both Skybox and indirect lighting)

vorg commented:
- [ ] rename `emmisive.glsl` -> `emissive-color.glsl`
- [ ] rename `encoding.glsl` -> `decode-encode.glsl`
- [ ] add source module to `fxaa`
- [ ] rename `normal` -> `normal`
- [ ] rename `workflow-roughness-metallic` -> `metallic-roughness`
- [ ] rename `workflow-specular-glossines` -> `specular-glossines`
- [ ] move chunks index from `index.js` to `chunks/index.js` ?


Other notes:
- we should probably remove displacement mapping as it doesn't recompute the normals and results in incorrect lighting unless we can fix it somehow

dmnsgn commented:
- Moved list of defines to NOTES.md for now
- Grouped `local_modules_old` and `glsl_old` (they both contained js and glsl) into `old`
- Renamed to kebab-case (which requires a `git move` so hopefully I didn't miss any)
- Separated back encode-decode, gamma and rgbm
- Flatten lights as well to stay consistent in the chunk folder
